Client/src/Pages/Incidents/index.jsx

```diff
@@ -102,7 +105,22 @@ const Incidents = () => {
   return (
     <>
       <Stack className="incidents" gap={theme.gap.large}>
-        {hasAnyIncidents ? (
+        {loading ? (
+          <>
+            <Stack direction="row" alignItems="center" gap={theme.gap.medium}>
+              <Skeleton variant="rounded" width={150} height={34} />
+              <Skeleton variant="rounded" width="15%" height={34} />
+              <Skeleton
+                variant="rounded"
+                width="20%"
+                height={34}
+                sx={{ ml: "auto" }}
+              />
+            </Stack>
+            <Skeleton variant="rounded" width="100%" height={300} />
+            <Skeleton variant="rounded" width="100%" height={100} />
```

Comment:
Could the whole Skeleton be extracted to a component within the class? Just to keep the main render method a bit more readable